src/solvers/smt2_incremental/smt_bit_vector_theory.cpp

```diff
@@ -4,6 +4,34 @@
 
 #include <util/invariant.h>
 
+const char *smt_bit_vector_theoryt::concatt::identifier()
+{
+  return "concat";
+}
+
+smt_sortt smt_bit_vector_theoryt::concatt::return_sort(
+  const smt_termt &lhs,
+  const smt_termt &rhs)
+{
+  const auto get_width = [](const smt_termt &term) {
+    return term.get_sort().cast<smt_bit_vector_sortt>()->bit_width();
+  };
+  return smt_bit_vector_sortt{get_width(lhs) + get_width(rhs)};
+}
+
+void smt_bit_vector_theoryt::concatt::validate(
+  const smt_termt &lhs,
+  const smt_termt &rhs)
+{
+  const auto lhs_sort = lhs.get_sort().cast<smt_bit_vector_sortt>();
```

Comment:
This appears to have the same structure as `validate_bit_vector_operator_arguments`, sans the `INVARIANT` for the same bit width of both the left and right operand.

Would it make sense to refactor this to be using a common procedure?

Reply:
Done.